src/otx/algo/instance_segmentation/maskrcnn_tv.py

```diff
@@ -5,140 +5,107 @@
-
-    from otx.core.metrics import MetricCallable
-    from otx.core.schedulers import LRSchedulerListCallable
-    from otx.core.types.label import LabelInfoTypes
-
-
-class _TVMaskRCNN(MaskRCNN):
-    """Torchvision MaskRCNN model with forward method accepting InstanceSegBatchDataEntity."""
-
-    def forward(self, entity: InstanceSegBatchDataEntity) -> dict[str, Tensor] | list[dict[str, Tensor]]:
-        """Overwrite GeneralizedRCNN forward method to accept InstanceSegBatchDataEntity."""
-        ori_shapes = [img_info.ori_shape for img_info in entity.imgs_info]
-        img_shapes = [img_info.img_shape for img_info in entity.imgs_info]
-
-        image_list = ImageList(entity.images, img_shapes)
-        targets = []
-        for bboxes, labels, masks, polygons in zip(
-            entity.bboxes,
-            entity.labels,
-            entity.masks,
-            entity.polygons,
-        ):
-            # NOTE: shift labels by 1 as 0 is reserved for background
-            _labels = labels + 1 if len(labels) else labels
-            targets.append(
-                {
-                    "boxes": bboxes,
-                    "labels": _labels,
-                    "masks": masks,
-                    "polygons": polygons,
-                },
-            )
 
-        features = self.backbone(image_list.tensors)
-        if isinstance(features, Tensor):
-            features = OrderedDict([("0", features)])
-        proposals, proposal_losses = self.rpn(image_list, features, targets)
+class TVMaskRCNN(ExplainableOTXInstanceSegModel):
+    """Implementation of torchvision MaskRCNN for instance segmentation."""
+
+    load_from: ClassVar[dict[str, Any]] = {"maskrcnn_resnet_50": MaskRCNN_ResNet50_FPN_V2_Weights.verify("DEFAULT")}
+    mean = (123.675, 116.28, 103.53)
+    std = (58.395, 57.12, 57.375)
+
+    AVAILABLE_MODEL_VERSIONS: ClassVar[list[str]] = [
+        "maskrcnn_resnet_50",
+    ]
+
+    def _create_model(self) -> nn.Module:
+        """Create MaskRCNN model with TV implementation."""
+        if self.model_name not in self.AVAILABLE_MODEL_VERSIONS:
+            msg = f"Model version {self.model_name} is not supported."
+            raise ValueError(msg)
+
+        # NOTE: Add 1 to num_classes to account for background class.
+        num_classes = self.label_info.num_classes + 1
+        weights = self.load_from[self.model_name]
+
+        # init model components, model itself and load weights
+        rpn_anchor_generator = _default_anchorgen()
+        backbone = MaskRCNNBackbone(model_name=self.model_name)
+        rpn_head = RPNHead(model_name=self.model_name, anchorgen=rpn_anchor_generator)
+        box_head = FastRCNNConvFCHead(model_name=self.model_name)
+        mask_head = MaskRCNNHeads(model_name=self.model_name)
 
-        detections, detector_losses = self.roi_heads(
-            features,
-            proposals,
-            image_list.image_sizes,
-            targets,
+        model = MaskRCNNTV(
```

Comment:
We should consider better naming conventions here, as we currently have `TV` used both as a prefix and a suffix in `TVMaskRCNN` and `MaskRCNNTV`, which makes it unclear what the differences are. Previously, the name `OTXTVMaskRCNN` was used (maybe?), likely to distinguish between OTX model module and the PyTorch TVMaskRCNN model module.

Reply:
I renamed it to MaskRCNNTV and named model from torchvision (in src/otx/algo/instance_segmentation/maskrcnn_tv.py) as simply MaskRCNN